### PR TITLE
Bug 1735711: lock down timing windows when deletes of the config object come in du…

### PR DESCRIPTION
**Pull request by gabemontero**
…ring samples upsert

@openshift/openshift-team-developer-experience fyi

@assign @adambkaplan

### CODE REVIEW

File: pkg/stub/handler.go [hunk 1]
```diff
@@ -854,14 +868,30 @@ func (h *Handler) setSampleManagedLabelToFalse(kind, name string) error {
...
+	cfg = h.refetchCfgMinimizeConflicts(cfg)
+	if cfg.DeletionTimestamp != nil {
+		return true
+	}
+	return false
+}
+
+// rc:  error - any api errors during upserts
+// rc:  bool - if we abort because we detected a delete
+func (h *Handler) createSamples(cfg *v1.Config, updateIfPresent, registryChanged bool, unskippedStreams, unskippedTemplates map[string]bool) (error, bool) {
```
Comment (adambkaplan) [inline pkg/stub/handler.go]: nit - golang convention is return error last
```suggestion
func (h *Handler) createSamples(cfg *v1.Config, updateIfPresent, registryChanged bool, unskippedStreams, unskippedTemplates map[string]bool) (bool, error) {
```

Comment (gabemontero) [reply pkg/stub/handler.go]: param order switched @adambkaplan - thanks !!

File: pkg/stub/handler.go [hunk 2]
```diff
@@ -724,7 +728,17 @@ func (h *Handler) Handle(event v1.Event) error {
 				return err
 			}
 
-			err = h.createSamples(cfg, true, registryChanged, unskippedStreams, unskippedTemplates)
+			abortForDelete, err := h.createSamples(cfg, true, registryChanged, unskippedStreams, unskippedTemplates)
+			if abortForDelete {
```
Comment (adambkaplan) [inline pkg/stub/handler.go]: We do this even if err != nil? Typical golang convention (and expectation) is that `err != nil` is handled first by method callers.

Comment (adambkaplan) [reply pkg/stub/handler.go]: /lgtm cancel

Comment (gabemontero) [reply pkg/stub/handler.go]: I would call this an exception to the rule, as the delete is blowing everything away, including whatever error tracking might have happened with any error processing

Comment (gabemontero) [reply pkg/stub/handler.go]: I can add a comment to that effect, assuming we can agree on this.

Comment (gabemontero) [reply pkg/stub/handler.go]: added clarifying comment @adambkaplan .... PTAL

**Comment by gabemontero** [comment]
flakes flakes flakes  posted one in particular on #4-dev-triage

/retest

**Pull request merged**